### DISCUSSION TITLE
Push hard on `primitives`

### DISCUSSION THREAD

**Discussion by tcharding** [rust-bitcoin/rust-bitcoin#3289] [Announcements]
Fellas, please consider modifying your workflow to prioritise review of `P-High` PRs. I am using `P-High` almost exclusively ATM to indicate "next step in creating `primitives`".

Link to PRs with filter enabled: https://github.com/rust-bitcoin/rust-bitcoin/pulls?q=is%3Aopen+is%3Apr+label%3AP-high

**Comment by tcharding** [comment]
I'll make an effort to check in on these PRs before I got to bed so I catch input from you when you come online @Kixunil

**Reply by tcharding** [reply]
I'm finding this a little hard because I'm often tired and cranky by the end of the day and @Kixunil's reviews, usually correct but often extremely particular, have a tendency to piss me off if I read them in the evening, I do appreciate them in the morning time when I'm not so sensitive :) In general I should be better, but so far I've not managed to be so I'm accepting my personal weakness and being more careful to not read reviews when I'm feeling prickly.

**Comment by tcharding** [comment]
> Unfortunately I have quite a lot going on lately and there's also many PRs, some of them more difficult to review.

ref: https://github.com/rust-bitcoin/rust-bitcoin/pull/3308#issuecomment-2333268442

@Kixunil it occurred to me that perhaps the best way to speed up work on `primitives` is for me to stop doing any other `rust-bitcoin` patches except things that push `primitives` forward. This conflicts with stabalizing `hashes` though. You review stuff in `secp` as well so I would not work there either but I could focus in `miniscript` and `bitcoid-json-rpc` since you don't review those as much, right?

I can't stop external contributors doing patches and you using your time to review them but I could request that you prioritize mine, would that be fair if I was only doing `primitives` stuff?

**Reply by Kixunil** [reply]
Yes, not doing unrelated things would help. But also trying to keep the PRs small (that way I can quickly check things before sleep) and also ordering required cleanups before large moves. Possibly not doing any cleanups of internal code while moving but we should keep in mind that while `internals` is unstable we probably don't want to bump its version way too often because that will make downstream consumers have duplicate dependencies more frequently.

What could also help is adding some markers to *PR titles* ("small:", "priority:"...) not just tags so that I can see them in notifications.

**Reply by Kixunil** [reply]
Also when doing moves to the smaller crates be as conservative as possible. If you only move the conversions and nothing else it's least likely that I object to something. :)

**Reply by tcharding** [reply]
Cool thanks man, please accept a standing apology for everytime I get snarky, this `primitives` work is really tedious but hard at the same time, and thanks for reviewing, I suppose it is tedious for you too.
